### ISSUE TITLE
no-this-assignment works weird (destructuring)

### ISSUE THREAD

**Issue by xahon** [palantir/tslint#4792]
### Bug Report

- __TSLint version__:     "tslint": "^5.18.0",
- __TypeScript version__:     "typescript": "3.5.3"
- __Running TSLint via__: Rider/WebStorm

#### TypeScript code being linted

```ts
class A {
    private a: number;
    private b: number;

    constructor() {
        const { b } = (this); // no error
        const { a } = this;   // error
    }
}
```

with `tslint.json` configuration:

```json
// You can modify the configuration here with your own.
// Make sure that your json is valid, and if you run into
// other issues, try checking the console for errors.

{
    "extends": ["tslint:recommended"],
    "rules": {
        "no-this-assignment": true
    } // add additional rules and their configuration
}
```

#### Actual behavior
error in 1 case

#### Expected behavior
no errors (destructuring instead of assignment)

**Comment by xahon** [comment]
I wonder why so obvious bugs still were not found and fixed

**Comment by adidahiya** [comment]
> I wonder why so obvious bugs were yet not found and fixed

FWIW, this is not a great tone for engaging in open source projects.

Also, you may want to check the rule documentation: https://palantir.github.io/tslint/rules/no-this-assignment/. In this case there is a rule option which should allow destructuring assignment.
